### PR TITLE
Improve Graph Snapshot for Benchmarks

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

This follows from #1123. It expands upon the (costly) setup step to greatly increase the number of entities, and ontological types. It also adds a `cargo make bench` command, and updates the README.

## 🔍 What does this change?

- Introduces a new cargo make task for benchmarking
- "Fixes" (current errors on) the test data by adding missing property types and such
- Improves the `seed` mechanism to call `update_` if `create_` fails because the thing already exists
- Changes seed_db to seed the database with a bunch of data. 

## 📜 Does this require a change to the docs?

- The README has been updated to say how the benchmarks can be ran

## ⚠️ Known issues

- The `seed_db` step takes quite a while. Perhaps we can export a prepared snapshot of the postgres database and load it at the start of the benchmarks. This would make the data somewhat non-introspectable and hard to update. In the meantime, we have chosen to **not** clean up the tables made by the benchmarks. Please read the warning in the updated README for more info.

## 🐾 Next steps

- Figure out a way to create a synthetic 'representative' test graph
- Improve the mechanism through which we express states of the graph, especially useful for tests and benchmarks

## 🛡 What tests cover this?

- The current (single) benchmark group

## ❓ How to test this?

1. Checkout the branch
2. Run `cargo make deployment-up`
3. Run `make recreate-db`
4. Run `make migrate-up`
5. Run `cargo bench`

### CODE REVIEW

File: packages/graph/hash_graph/bench/benches/representative_read/seed.rs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+pub async fn setup_and_extract_samples(store_wrapper: &mut StoreWrapper) -> Samples {
+    // TODO: We'll want to test distribution across accounts
+    //  https://app.asana.com/0/1200211978612931/1203071961523000/f
+    // We use a hard-coded UUID to keep it consistent across tests so that we can use it as a
+    // parameter argument to criterion and get comparison analysis
+    let account_id =
+        AccountId::new(Uuid::from_str("d4e16033-c281-4cde-aa35-9085bf2e7579").unwrap());
+
+    let not_empty: bool = store_wrapper
```
Comment (TimDiekmann) [inline packages/graph/hash_graph/bench/benches/representative_read/seed.rs:229]:
```suggestion
    let account_id_exists: bool = store_wrapper
```

File: packages/graph/hash_graph/bench/benches/representative_read/seed.rs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            r#"
+            SELECT EXISTS(SELECT 1 FROM accounts WHERE account_id=$1)
+            "#,
+            &[&account_id],
+        )
+        .await
+        .expect("failed to check if account id exists")
+        .get(0);
+
+    if !(not_empty) {
```
Comment (TimDiekmann) [inline packages/graph/hash_graph/bench/benches/representative_read/seed.rs:242]:
```suggestion
    if !account_id_exists {
```

File: packages/graph/hash_graph/bench/benches/util.rs [hunk 3]
```diff
@@ -56,32 +56,54 @@ impl StoreWrapper {
...
-            client
-                .execute(
+            let exists: bool = client
+                .query_one(
                     r#"
-                    /* KILL ALL EXISTING CONNECTION FROM ORIGINAL DB*/
-                    SELECT pg_terminate_backend(pg_stat_activity.pid) FROM pg_stat_activity
-                    WHERE pg_stat_activity.datname = $1 AND pid <> pg_backend_pid();
+                    SELECT EXISTS(
+                     SELECT datname FROM pg_catalog.pg_database WHERE datname = $1
```
Comment (TimDiekmann) [inline packages/graph/hash_graph/bench/benches/util.rs:63]: Either
```suggestion
                        SELECT 1 FROM pg_catalog.pg_database WHERE datname = $1
```
or
```suggestion
                    SELECT 1 FROM pg_catalog.pg_database WHERE datname = $1
```

File: packages/graph/hash_graph/bench/benches/util.rs [hunk 4]
```diff
@@ -56,32 +56,54 @@ impl StoreWrapper {
...
-                    &format!(
+                .expect("failed to check if database exists")
+                .get(0);
+
+            if fail_on_exists && exists {
+                panic!(
+                    "database `{}` exists, and fails_on_exists was set to true",
+                    bench_db_name
+                );
+            }
```
Comment (TimDiekmann) [inline packages/graph/hash_graph/bench/benches/util.rs:77]:
```suggestion
            assert!(
                !exists || !fail_on_exists,
                "database `{}` exists, and `fails_on_exists` was set to `true`",
                bench_db_name,
            );
```

**Review by TimDiekmann — CHANGES_REQUESTED** [review]
A few nits, otherwise looking good and it's working locally 🎉

File: packages/graph/hash_graph/bench/benches/util.rs (hunk 4 above)
Comment (Alfred-Mountfield) [reply packages/graph/hash_graph/bench/benches/util.rs]: Have added something similar, thanks

File: packages/graph/hash_graph/bench/benches/util.rs (hunk 3 above)
Comment (Alfred-Mountfield) [reply packages/graph/hash_graph/bench/benches/util.rs]: Added the first one, thanks

File: packages/graph/hash_graph/bench/benches/representative_read/seed.rs (hunk 2 above)
Comment (Alfred-Mountfield) [reply packages/graph/hash_graph/bench/benches/representative_read/seed.rs]: Chose something similar, and added a comment

**Review by TimDiekmann — APPROVED** [review]

**Pull request merged**